src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -759,6 +759,24 @@ IEnumerator IEnumerable.GetEnumerator()
             return new Enumerator(this, Enumerator.KeyValuePair);
         }
 
+        /// <summary>
+        /// Ensures that the capacity of this dictionary is at least the given minimum value
+        /// </summary>
+        /// <param name="capacity"></param>
+        /// <returns></returns>
+        public int EnsureCapacity(int capacity)
+        {
+            if (capacity < 0)
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.capacity);
+            if (_entries == null)
+                Initialize(0);
+            if (capacity <= _entries.Length)
+                return _entries.Length;
+            int newSize = HashHelpers.ExpandPrime(capacity);
+            Resize(newSize, false);
```

Comment:
code style, please include the param name for opaque booleans ie
`Resize(newSize, forceNewHashCodes: false)`